agent/src/beerocks/slave/link_metrics/ieee802_3_link_metrics_collector.cpp

```diff
@@ -0,0 +1,246 @@
+             * This message contains the stats for the interface we are interested in
+             */
+            if (0 == strcmp(local_interface_name.c_str(), (char *)RTA_DATA(attribute))) {
+                result = true;
+            }
+            break;
+        case IFLA_STATS:
+            if (result) {
+                struct rtnl_link_stats *stats = (struct rtnl_link_stats *)RTA_DATA(attribute);
+
+                /**
+                 * Get interface speed and set PHY rate accordingly.
+                 * Speed values other than 100Mbps and 1Gbps are ignored.
+                 */
+                uint16_t phy_rate = UINT16_MAX;
+                uint32_t speed    = 0;
+                if (beerocks::net::network_utils::linux_iface_get_speed(local_interface_name,
+                                                                        speed)) {
+                    if (SPEED_100 == speed) {
+                        phy_rate = 100;
+                    } else if (SPEED_1000 == speed) {
+                        phy_rate = 1000;
+                    }
+                }
+
+                link_metrics.transmitter.packet_errors       = stats->tx_errors;
+                link_metrics.transmitter.transmitted_packets = stats->tx_packets;
+                /**
+                 * Note: The MAC throughput capacity is a function of the physical data rate and
+                 * of the MAC overhead. We could somehow compute such overhead or, for simplicity,
+                 * set the MAC throughput as a percentage of the physical data rate. To make
+                 * things even simpler, we will set the MAC throughput capacity with the same
+                 * value as the physical data rate, as if there was no overhead at all.
+                 */
+                link_metrics.transmitter.mac_throughput_capacity = phy_rate;
+                // Note: For simplicity, link availability is set to "100% of the time"
+                link_metrics.transmitter.link_availability = 100;
+                link_metrics.transmitter.phy_rate          = phy_rate;
+
+                link_metrics.receiver.packet_errors    = stats->rx_errors;
+                link_metrics.receiver.packets_received = stats->rx_packets;
+                link_metrics.receiver.rssi             = UINT8_MAX;
+            }
+            break;
+        }
+    }
+
+    return result;
+}
+
+static bool get_link_metrics(int fd, const std::string &local_interface_name,
+                             sLinkMetrics &link_metrics)
+{
+    bool result = false;
+
+    struct sockaddr_nl kernel; /* the remote (kernel space) side of the communication */
+
+    struct msghdr rtnl_msg; /* generic msghdr struct for use with sendmsg */
+    struct iovec io;        /* IO vector for sendmsg */
+    struct nl_req_t req;    /* structure that describes the Netlink packet itself */
+
+    /**
+     * Netlink socket is ready for use, prepare and send request
+     */
+    memset(&rtnl_msg, 0, sizeof(rtnl_msg));
```

Comment:
Although `memset` is not banned, it is recommended to use `memset_s` or an equivalent c++ STD function such as `std::fill`. See https://github.com/intel/safestringlib/wiki/SDL-List-of-Banned-Functions

Reply:
I've set the initial value of structs to zero on declaration and removed all calls to `memset`. 
Looks like `memset_s` is not available.